mmv1/third_party/terraform/services/servicenetworking/resource_service_networking_connection_test.go

```diff
@@ -62,6 +63,34 @@ func TestAccServiceNetworkingConnection_abandon(t *testing.T) {
 	})
 }
 
+func TestAccServiceNetworkingConnection_updateOnCreationFail(t *testing.T) {
+	t.Parallel()
+
+	network := fmt.Sprintf("tf-test-service-networking-connection-update-%s", acctest.RandString(t, 10))
+	addr1 := fmt.Sprintf("tf-test-%s", acctest.RandString(t, 10))
+	addr2 := fmt.Sprintf("tf-test-%s", acctest.RandString(t, 10))
+	service := "servicenetworking.googleapis.com"
+	org_id := envvar.GetTestOrgFromEnv(t)
+	billing_account := envvar.GetTestBillingAccountFromEnv(t)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testServiceNetworkingConnectionDestroy(t, service, network),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccServiceNetworkingConnections(network, addr1, addr2, "servicenetworking.googleapis.com", org_id, billing_account, false),
+				ExpectError: regexp.MustCompile("Cannot modify allocated ranges in CreateConnection. Please use UpdateConnection"),
+			},
+			{
+				Config:             testAccServiceNetworkingConnections(network, addr1, addr2, "servicenetworking.googleapis.com", org_id, billing_account, true),
+				ExpectNonEmptyPlan: true,
```

Comment:
why do we expect a non-empty plan here again?

Reply:
Actually, `google_service_networking_connection.foobar` and `google_service_networking_connection.foobar1` are the same service networking connection as they have the same network and service.  When `update_on_creation_fail` is set to true, the service networking connection is updated with  peering range `google_compute_global_address.foobar1` successfully through `google_service_networking_connection.foobar1`. After refresh, `google_service_networking_connection.foobar` will also have the peering range `google_compute_global_address.foobar1`. The plan is not empty.